### ISSUE TITLE
Insights is sometimes picking up old local authority codes, leading to duplicates.

### ISSUE THREAD

**Issue by drkane** [ThreeSixtyGiving/grantvis#161]
I think there's an inconsistency in the source data ([from drkane/geo-lookups](https://github.com/drkane/geo-lookups)) where some records have a `ladcd_active` which should be used as the latest local authority code, and some only have a `ladcd` which can contain old defunct codes

This creates duplicates in the Insights local authority chart, eg here where there are two entries for Glasgow City (the one at the top is for the latest code `S12000049` and the one at the bottom is for the old code `S12000046`).

![Screenshot 2022-11-30 at 15-13-09 360Giving Insights](https://user-images.githubusercontent.com/1049910/204835561-812a4a35-1ffd-4c50-871a-38d25ee8b772.png)

This is caused by two different area types used in different grants.

**Grant A locationLookup**

So we have one grant which has a beneficiary location of `S12000046` which is the old local authority code. This gets data from the [geo-lookups/area-all-codes.csv](https://github.com/drkane/geo-lookups#area_all_codescsv) file, which uses `ladcd` instead of `ladcd_active`:

```json
{
    "areacode": "S12000046",
    "areaname": "Glasgow City",
    "areatype": "la", <-lookup based on local authority
    "ladcd": "S12000049",  <-new code as `ladcd`
    "source": "beneficiaryLocation",
    "sourceCode": "S12000046",  <-old code as lookup
    ...
}
```

**Grant B locationLookup**

The other grant uses a ward code, which gives the new code as `ladcd_active` and the old code as `ladcd`.

```json
{
    "areacode": "S13002658",
    "areaname": "East Centre",
    "areatype": "ward", <-lookup based on ward
    "ladcd": "S12000046", <-old code as `ladcd`
    "ladcd_active": "S12000049", <-new code as `ladcd_active`
    "source": "beneficiaryLocation",
    "sourceCode": "S13002658", <-ward code as lookup
}
```

I think the solution is in two parts:

1) I add a `ladcd_active` field to the [geo-lookups/area-all-codes.csv](https://github.com/drkane/geo-lookups#area_all_codescsv) CSV, which has the same contents as `ladcd` (I'll do the same for `ladnm` > `ladnm_active`.

2) Insights changes the data load process found here:

https://github.com/ThreeSixtyGiving/360insights/blob/5e845b3e3cf7c7164b29bd95ff2ab5dc2531fc3f/insights/commands/fetch_data.py#L221

to instead read

```
g.additional_data->'locationLookup'->0->>'ladcd_active' as "insights_geo_la",
```

or for belt and braces

```
COALESCE(
    g.additional_data->'locationLookup'->0->>'ladcd_active',
    g.additional_data->'locationLookup'->0->>'ladcd'
)  as "insights_geo_la",
```

**Comment by mariongalley** [comment]
@drkane Does this need fixing by ODSC or you?

**Comment by KDuerden** [comment]
I think this issue is the cause of an anomaly that a funder has just reported for their data in Insights. 

https://insights.threesixtygiving.org/data?funders=360G-WilliamGrantFdn
Over 100 grants appear in 'Glasgow City' but only 1 in the zoomed in map view, unless you select [Glasgow from the filters](https://insights.threesixtygiving.org/data?funders=360G-WilliamGrantFdn&localAuthorities=S12000046)

![image](https://user-images.githubusercontent.com/20462640/220415549-f41a4712-f8e5-437b-b1a4-95bb18f43c3d.png)

This seems to be because most of their grants have the old Glasgow City geocode (S12000046) and only two have the current one S12000049. Does this mean only the current geocodes get visualised by the map by default?

I couldn't work out why their data has been matched with the old geocode, because the postcodes and Ward level geocodes they share are all up to date, and when I check them in Find that Postcode they are related to the current rather than the old Glasgow area. I found this problem affecting lots of other data.

GrantNav download for all grants in the Glasgow filter has 6597 grants with current S12000049 geocode. 

There are 5531 grants under the inactive S12000046 geocode. Although 3893 have the old geocode shared as their only form of geodata, and some spot checks of beneficiary location data I have found some Ward codes that are also inactive and are linked to the old geocode. However it doesn't explain all of the linkages - especially when the only geodata shared is a current postcode.

**Comment by mariongalley** [comment]
@michaelwood Is David's comment above clear, and can this be tackled as part of iterative improvements, or is there more needed from David?

**Comment by mariongalley** [comment]
@drkane 

> I think the solution is in two parts:

Have you done point 1 or is that still to do?

**Comment by drkane** [comment]
Yep - this bit has been done:

> I add a ladcd_active field to the geo-lookups/area-all-codes.csv CSV, which has the same contents as ladcd (I'll do the same for ladnm > ladnm_active.

**Comment by mariongalley** [comment]
This issue is fixed in the new iteration of Insights PR #208
